ponys.js

```diff
@@ -15,18 +15,16 @@ export default class {
 			template = templateElement;
 		}
 		template = template.content;
-		url = new URL(url, location.href);
+		url = new URL(url, location.href.startsWith('about:') ? document.baseURI : location.href);
```

Comment:
Do you see any disadvantage to just using `document.baseURI` in all cases here? It seems it will match `location.href` in all cases where the document does not have a separate `<base>` specified, and perhaps (given that this relative URL rewriting is already a bit of a hack) it would be good if the import scripts would respect such a base URL, given that it's user-supplied in the same way as an import map would be...? :thinking:

```suggestion
		url = new URL(url, document.baseURI);
```

Reply:
I thought the same thing, but I did not experiment enough to determine any edges cases, if any. My main thought process was just to make it backwards compatabile, so if there are any differences, we preserve existing behavior except for this specific scenario (when using srcdoc or other `about:` scenarios).

Additionally since this logic is really only needed for the relative import logic, which might not even be touched in most use cases, we could probably defer its execution until actually needed.